### PR TITLE
Add ability to batch-process all P223's available on seattleschools.org

### PR DESCRIPTION
**Pull request by abrauninger**
This PR adds `p223_pdf_batch.py`, which runs the existing code in `p223_pdf_to_csv.py` for many P223 PDFs at once.

This PR also does the following:
* Commit all P223 PDFs currently available on seattleschools.org (September 2019 through September 2024).
* Commit output CSVs: One CSV per month and a single `all.csv` that concatenates data from all P223s.
  * Committing the PDFs and CSVs adds about 20 MB of files to this repo. 
* Fix bugs in `p223_pdf_to_csv.py`:
  * Handle `State FDK` grades, which is how SPS used to refer to kindergarten in older P223's (~2021 and before)
  * Fix typo in column header (`P223 Total Coun` -> `P223 Total Count`)
* Flesh out `README.md` with instructions for accessing the extracted P223 data as well as running the Python scripts.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,44 @@
...
+
+#### Setup
+
+**Note:** This has only been tested on macOS.
+
+```console
+$ brew install pdftotext
+```
+
+`tr` is also required, but `tr` should already be installed by the OS:
```
Comment (awong-dev) [inline README.md:14]: yay unix

File: extractors/p223_pdf_batch.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    return month
+
+
+def main(input_directory, output_directory):
+    pdftotext = shutil.which('pdftotext')
+    if pdftotext is None:
+        sys.exit("'pdftotext' is not installed. See installation instructions in README.md.")
+
+    tr = shutil.which('tr')
```
Comment (awong-dev) [inline extractors/p223_pdf_batch.py:44]: come to think of it, we could kill the tr -s dependency by just running all lines through

re.sub('\s+', ' ')

File: extractors/p223_pdf_batch.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    for pdf_path in pdf_paths:
+        with tempfile.NamedTemporaryFile(delete_on_close=False) as squished_tempfile:
+            # '-f 2' to start processing on the second page of the PDF (since page 1 is a cover page).
+            pdftotext_process = subprocess.Popen([pdftotext, '-layout', pdf_path, '-f', '2', '-'], stdout=subprocess.PIPE)
+            tr_process = subprocess.Popen([tr, '-s', "' '"], stdin=pdftotext_process.stdout, stdout=squished_tempfile)
+            pdftotext_process.stdout.close()
+            out, err = tr_process.communicate()
+
+            if err is not None:
+                sys.exit(f"'pdftotext' and/or 'tr' had an error:\n{err}")
```
Comment (awong-dev) [inline extractors/p223_pdf_batch.py:61]: raise RuntimeError() instead of sys.exit()?

File: extractors/p223_pdf_batch.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+            if out is not None:
+                print(f"'pdftotext' and/or 'tr' produced output on stdout, which was unexpected:\n{out}")
+
+            squished_tempfile.close()
```
Comment (awong-dev) [inline extractors/p223_pdf_batch.py:66]: This is a little unexpected in a `with` clause.  Does .flush() work?  Or maybe just even pass in the file handle and modify the lower script to accept it?

**Review by awong-dev — COMMENTED** [review]
Adding comments, but I'm happy to fix those myself.

Overall looks good except for the data file submits... I think we should keep those out of the source repo.

**Review by awong-dev — APPROVED** [review]

File: extractors/p223_pdf_batch.py (hunk 4 above)
Comment (abrauninger) [reply extractors/p223_pdf_batch.py:66]: I'm following the example at https://docs.python.org/3/library/tempfile.html#examples:

```python
# create a temporary file using a context manager
# close the file, use the name to open the file again
with tempfile.NamedTemporaryFile(delete_on_close=False) as fp:
    fp.write(b'Hello world!')
    fp.close()
# the file is closed, but not removed
# open the file again by using its name
    with open(fp.name, mode='rb') as f:
        f.read()
b'Hello world!'

# file is now removed
```

Comment (awong-dev) [reply extractors/p223_pdf_batch.py:66]: wow...okay. That makes some weird sense but still feels just awkward.  Thanks for the pointer!

**Pull request merged**